### PR TITLE
Add documentation for default-language field

### PR DESCRIPTION
**Pull request by VeryMilkyJoe**
---
Please include the following checklist in your PR:

* [ ] Patches conform to the [coding conventions](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#other-conventions).
* [ ] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#changelog).
* [ ] The documentation has been updated, if necessary.
* [ ] Include [manual QA notes](https://github.com/haskell/cabal/blob/master/CONTRIBUTING.md#qa-notes) if your PR relates to cabal-install.

Bonus points for added automated tests!

### CODE REVIEW

File: doc/cabal-package.rst [hunk 1]
```diff
@@ -1858,7 +1858,13 @@ system-dependent values for these fields.
...
    :since: 1.12
 
-   TBW
+    Specifies a group of language extensions to be activated for the project, see `here <https://downloads.haskell.org/ghc/latest/docs/users_guide/exts/control.html#controlling-extensions>` for details.
+
+    The possible values are:
+
+    -  ``Haskell2010``
+    -  ``Haskell98``
+    -  ``GHC2021``
```
Comment (ffaf1) [inline doc/cabal-package.rst:1867]: Is there a reason not to have this list “chronologically” sorted (98/2010/2021 or reversed)?

**Review by ffaf1 — APPROVED** [review]
Improving docs is always welcome!

File: doc/cabal-package.rst (hunk 1 above)
Comment (VeryMilkyJoe) [reply doc/cabal-package.rst:1867]: not at all, i think newest to oldest makes the most sense?

Comment (ffaf1) [reply doc/cabal-package.rst]: Fine by me!

**Review by Kleidukos — APPROVED** [review]

**Comment by Kleidukos** [comment]
@VeryMilkyJoe Thank you very much for this contribution, it is highly appreciated!

**Comment by VeryMilkyJoe** [comment]
> Blocking for the time being (until markup is fixed).

I've fixed the link, thanks for noticing!

**Comment by ulysses4ever** [comment]
Cool, thank you! We'll try to merge it soon.

**Comment by ulysses4ever** [comment]
@mergify rebase

**Review by ulysses4ever — APPROVED** [review]

**Comment by ulysses4ever** [comment]
@mergify rebase

**Pull request merged**